hudi-common/src/main/java/org/apache/hudi/common/table/read/DeleteContext.java

```diff
@@ -94,15 +95,19 @@ private static Option<Pair<String, String>> getCustomDeleteMarkerKeyValue(Proper
    * @param schema table schema to check
    * @return whether built-in delete field is included in the table schema
    */
-  private static boolean hasBuiltInDeleteField(Schema schema) {
-    return schema.getType() != Schema.Type.NULL && schema.getField(HOODIE_IS_DELETED_FIELD) != null;
+  private static boolean hasBuiltInDeleteField(HoodieSchema schema) {
+    return schema.getType() != HoodieSchemaType.NULL && schema.getField(HOODIE_IS_DELETED_FIELD).isPresent();
   }
 
   /**
    * Returns position of hoodie operation meta field in the schema
    */
-  private static int getHoodieOperationPos(Schema schema) {
-    return Option.ofNullable(schema.getField(HoodieRecord.OPERATION_METADATA_FIELD)).map(Schema.Field::pos).orElse(-1);
+  private static int getHoodieOperationPos(HoodieSchema schema) {
+    return Option.ofNullable(schema.getField(HoodieRecord.OPERATION_METADATA_FIELD))
```

Comment:
We don't need the initial `Option.ofNullable` here since the `getField` returns an option now instead of a null value